modules/apps/portal-security/portal-security-permission-test/src/testIntegration/java/com/liferay/portal/security/permission/test/ResourceActionsTest.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.security.permission.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.portal.kernel.security.permission.ResourceActionsUtil;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import java.util.Locale;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Lianne Louie
+ */
+@RunWith(Arquillian.class)
+public class ResourceActionsTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayIntegrationTestRule liferayIntegrationTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testGetModelResourceFromUnsupportedLanguage() {
+		String actual = ResourceActionsUtil.getModelResource(
```

Comment:
@wanderlast I do not understand how your test reflects the issue described in the ticket.

Reply:
@tinatian the getModelResource() method is how we retrieve relevant language strings in a number of locations across the portal such as when we fill the Asset Pubisher's possible source list in asset-publisher-web's source.jsp (among other locations). When this method is specifically used, it tries to find the relevant value for its key in two areas: it checks against its module's resource bundle for the key-value pair within the given language using LanguageUtil.get() and it checks against the aggregate bundle of all language keys for the given language. This presents an issue when the language key we need is both unsupported and in a different module.

For example, using our AP example again, we are using getModelResource() to retrieve the model resource names of different portlets such as model.resource.com.liferay.dynamic.data.lists.model.DDLRecord. This takes place in asset-publisher-web and, therefore, LanguageUtil.get() does not have knowledge of these keys since they're typically stored in the relevant web module. Therefore, we check against the aggregate bundle which does find the correct value in dynamic-data-lists-web's Language_en_GB.properties file. However, since this language is only partially supported, the k-v pair it has is: model.resource.com.liferay.dynamic.data.lists.model.DDLRecord=Dynamic Data Lists Record (Automatic Copy). It is autogenerated as such by our language builder, but it is never stripped of its "Automatic Copy" notation since it bypasses our two safeguards against this behavior: we do not support the language on our translation service, CrowdIn, and therefore the key-value is overlooked/never properly stripped of the notation and we are not using LanguageUtil.get() which naturally has a call to fixValues() that would strip this notation away regardless. When retrieving from the aggregate bundle, without the fix in this PR the value is returned as-is and therefore includes the "(Automatic X)" notation. The test is meant to simulate these conditions: it checks for the value of a key against the unsupported/beta en-GB and then checks against en-US. The resulting values should theoretically be equal for the chosen key since both languages are English and the chosen key-value pair in this case should not have any changes due to locale differences between UK English vs US English. If it isn't, then we know that this issue has gone wrong and therefore the test fails.